java-vtl-script/src/test/java/no/ssb/vtl/script/functions/VTLSubstrTest.java

```diff
@@ -0,0 +1,165 @@
+package no.ssb.vtl.script.functions;
+
+/*
+ * ========================LICENSE_START=================================
+ * Java VTL
+ * %%
+ * Copyright (C) 2016 - 2017 Pawel Buczek
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =========================LICENSE_END==================================
+ */
+
+import no.ssb.vtl.model.VTLNumber;
+import no.ssb.vtl.model.VTLObject;
+import no.ssb.vtl.model.VTLString;
+import org.assertj.core.util.Lists;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+
+public class VTLSubstrTest implements VTLNumberFunctionTest {
```

Comment:
Try to only test one concept pr test method which means you will (in general) have only one `invoke` of the `VTLSubstr` function pr test method, and then give the test a name explaining the concept it tests (e.g. `testStartPositionGreaterThanStringLength()`)

Reply:
This is what I usually do. But if you look at the test classes in no.ssb.vtl.script.functions, all of them are written like that and I think it's good to follow the same pattern.

Maybe I should tweak the `VTLNumberFunctionTest` so that it does not point only to number functions.

Reply:
Ah! I see what you have done. You have implemented the `VTLNumberFunctionTest` interface. But `VTLSubstr` is not a number function, it is a string function and thus the pattern does not apply!

Reply:
Yes. Please take a look at f5de9eb that should fix it.